src/algorithms/tonal/pitchyinprobabilitieshmm.h

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2006-2018  Music Technology Group - Universitat Pompeu Fabra
+ *
+ * This file is part of Essentia
+ *
+ * Essentia is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU Affero General Public License as published by the Free
+ * Software Foundation (FSF), either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the Affero GNU General Public License
+ * version 3 along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+
+#ifndef ESSENTIA_PITCHYINPROBABILITIESHMM_H
+#define ESSENTIA_PITCHYINPROBABILITIESHMM_H
+
+#include "algorithmfactory.h"
+
+namespace essentia {
+namespace standard {
+
+class PitchYinProbabilitiesHMM : public Algorithm {
+
+ private:
+  Input<std::vector<std::vector<Real> > > _pitchCandidates;
+  Input<std::vector<std::vector<Real> > > _probabilities;
+  Output<std::vector<Real> > _pitch;
+
+  Algorithm* _viterbi;
+
+  Real _minFreq;
+  size_t _nBPS;
+  Real _selfTrans;
+  Real _yinTrust;
+  size_t _nPitch;
+  size_t _transitionWidth;
+  std::vector<Real> _freqs;
+
+  std::vector<Real> _init;
+  std::vector<size_t> _from;
+  std::vector<size_t> _to;
+  std::vector<Real> _transProb;
+
+  std::vector<Real> _tempPitch;
+
+ public:
+  PitchYinProbabilitiesHMM() {
+    declareInput(_pitchCandidates, "pitchCandidates", "the pitch candidates");
+    declareInput(_probabilities, "probabilities", "the pitch probabilities");
+    declareOutput(_pitch, "pitch", "pitch frequencies in Hz");
+
+    _viterbi = AlgorithmFactory::create("Viterbi");
+  }
+
+  void declareParameters() {
+    declareParameter("minFreq", "minimum detected frequency", "(0,inf)", 61.735);
+    declareParameter("nBPS", "number of bins per semitone", "(1,inf)", 5);
```

Comment:
Maybe use a ```binResolution``` parameter for consistency with parameters in [PitchSalienceFunction](https://essentia.upf.edu/documentation/reference/std_PitchSalienceFunction.html)

Reply:
changed to numberBinsPerSemitone